src/internals.jl

```diff
@@ -38,14 +38,22 @@ function validate_input(iVecchiaMLE::VecchiaMLEInput)
         @assert is_csc_format(iVecchiaMLE) "rowsL and colsL are not in CSC format"
     end
 
-    if !isnothing(iVecchiaMLE.lvar_diag)    
+    if !isnothing(iVecchiaMLE.lvar_diag)
+        # Check that lower bounds are positive
+        @assert mapreduce(x -> x > 0.0, &, iVecchiaMLE.lvar_diag) "User given lvar_diag must have all positive entries"
         @assert_eq length(iVecchiaMLE.lvar_diag) cache.n
     end
     
     if !isnothing(iVecchiaMLE.uvar_diag)
+        @assert mapreduce(x -> x > 0.0, &, iVecchiaMLE.uvar_diag) "User given uvar_diag must have all positive entries"
         @assert_eq length(iVecchiaMLE.uvar_diag) cache.n 
     end
     
+    # Check if lvar_diag .< iVecchiaMLE.uvar_diag
+    if !isnothing(iVecchiaMLE.lvar_diag) && !isnothing(iVecchiaMLE.uvar_diag)
+        @assert mapreduce(x -> x < 0.0, &, iVecchiaMLE.lvar_diag .- iVecchiaMLE.uvar_diag) "lvar_diag must be .< uvar_diag"
```

Comment:
```suggestion
        @assert mapreduce(<, &, iVecchiaMLE.lvar_diag, iVecchiaMLE.uvar_diag) "lvar_diag must be .< uvar_diag"
```

Reply:
`iVecchiaMLE.lvar_diag .- iVecchiaMLE.uvar_diag` will allocate an array. The `.-` does nothing here.